source/Coop/Mod/Managers/ServerManager.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Collections.Generic;
+using Coop.Mod.DebugUtil;
+using Coop.Mod.Serializers;
+using SandBox;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.Core;
+using TaleWorlds.SaveSystem.Load;
+
+namespace Coop.Mod.Managers
+{
+    public class ServerGameManager : CampaignGameManager
+    {
+        public ServerGameManager() : base() { }
+        public ServerGameManager(LoadResult saveGameData) : base(saveGameData) { }
+
+        ~ServerGameManager()
+        {
+            // TODO save all heros
+        }
+
+        public override void OnLoadFinished()
+        {
+            base.OnLoadFinished();
+            CLICommands.StartServer(new List<string>());
+            CoopClient.Instance.RemoteStoreCreated += (remoteStore) => {
+                remoteStore.OnObjectReceived += (objId, obj) =>
+                {
+                    if (obj is PlayerHeroSerializer serializedPlayerHero)
+                    {
+                        Hero hero = (Hero)serializedPlayerHero.Deserialize();

```

Comment:
Might want to consider removing the `obj` from the store. It will not be automatically removed!